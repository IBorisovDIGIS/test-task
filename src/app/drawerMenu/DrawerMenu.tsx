import React from 'react';
import clsx from 'clsx';
import {
  Divider, Drawer, IconButton, List, makeStyles,
} from '@material-ui/core';
import logo from '../../assets/icons/logo.svg';
import { drawerWidth, menuItems } from '../../constants';
import MenuItem from './components/MenuItem';
import { black } from '../../constants/colors';
import CustomIcon from '../common/CustomIcon';

const useStyles = makeStyles((theme) => ({
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    backgroundColor: black,
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  imageIcon: {
    width: 27,
    height: 40,
  },
  iconRoot: {
    textAlign: 'center',
    width: '100%',
    height: '100%',
  },
}));

interface IDrawerMenu {
    open: boolean;
    handleDrawerToggle: () => void;
}

const DrawerMenu: React.FC<IDrawerMenu> = ({ open, handleDrawerToggle }: IDrawerMenu) => {
  const classes = useStyles();

  return (
    <Drawer
      variant="permanent"
      classes={{
        paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
      }}
      open={open}
    >
      <div className={classes.toolbarIcon}>
        <IconButton onClick={handleDrawerToggle} id="menu-button">
          <CustomIcon icon={logo} iconClasses={classes.imageIcon} alt="logo" />
        </IconButton>
      </div>
      <Divider />
      <List>
        {menuItems.map((item) => <MenuItem {...item} key={item.label} />)}
      </List>
    </Drawer>
  );
};

export default DrawerMenu;
