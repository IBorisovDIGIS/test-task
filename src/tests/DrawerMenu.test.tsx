import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DrawerMenu from '../app/drawerMenu/DrawerMenu';

configure({ adapter: new Adapter() });

describe('Drawer menu should render correctly', () => {
  const handleDrawerToggle = jest.fn();

  it('Should renders correctly', () => {
    const container = shallow(<DrawerMenu open={false} handleDrawerToggle={handleDrawerToggle} />);

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should open menu correctly', () => {
    const closedMenu = shallow(<DrawerMenu open={false} handleDrawerToggle={handleDrawerToggle} />);
    const openMenu = shallow(<DrawerMenu open={true} handleDrawerToggle={handleDrawerToggle} />);
    closedMenu.setProps({ open: true });
    expect(closedMenu).toEqual(openMenu);
  });

  it('Should call handleDrawerToggle on logo button press', () => {
    const container = shallow(<DrawerMenu open={false} handleDrawerToggle={handleDrawerToggle} />);

    container.find('#menu-button').simulate('click');
    expect(handleDrawerToggle).toHaveBeenCalledTimes(1);
  });
});

