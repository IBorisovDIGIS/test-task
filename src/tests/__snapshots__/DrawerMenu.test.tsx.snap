// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer menu should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-2 makeStyles-drawerPaperClose-3",
    }
  }
  open={false}
  variant="permanent"
>
  <div
    className="makeStyles-toolbarIcon-1"
  >
    <WithStyles(ForwardRef(IconButton))
      id="menu-button"
      onClick={[MockFunction]}
    >
      <CustomIcon
        alt="logo"
        icon="logo.svg"
        iconClasses="makeStyles-imageIcon-4"
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <MenuItem
      icon="menu.svg"
      key="Dashboard"
      label="Dashboard"
    />
    <MenuItem
      icon="menu.svg"
      key="Services"
      label="Services"
    />
    <MenuItem
      icon="menu.svg"
      key="Account"
      label="Account"
    />
    <MenuItem
      icon="menu.svg"
      key="Messages"
      label="Messages"
    />
    <MenuItem
      icon="menu.svg"
      key="Test"
      label="Test"
    />
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Drawer))>
`;
